iceoryx_hoofs/include/iceoryx_hoofs/log/building_blocks/console_logger.hpp

```diff
@@ -64,35 +70,39 @@ class ConsoleLogger
 
     void assumeFlushed() noexcept;
 
+    // AXIVION Next Construct AutosarC++19_03-A3.9.1 : Not used as an integer but a low-level C-style string
     void logString(const char* message) noexcept;
 
     void logBool(const bool value) noexcept;
 
-    template <typename T, typename std::enable_if_t<std::is_arithmetic<T>::value, int> = 0>
-    void logDec(const T val) noexcept;
+    template <typename T, typename std::enable_if_t<std::is_arithmetic<T>::value, bool> = 0>
+    void logDec(const T value) noexcept;
 
     template <typename T,
               typename std::enable_if_t<(std::is_integral<T>::value && std::is_unsigned<T>::value)
-                                            || std::is_floating_point<T>::value,
-                                        int> = 0>
-    void logHex(const T val) noexcept;
+                                            || std::is_floating_point<T>::value || std::is_pointer<T>::value,
+                                        bool> = 0>
+    void logHex(const T value) noexcept;
 
-    template <typename T, typename std::enable_if_t<std::is_integral<T>::value && std::is_unsigned<T>::value, int> = 0>
-    void logOct(const T val) noexcept;
+    template <typename T, typename std::enable_if_t<std::is_integral<T>::value && std::is_unsigned<T>::value, bool> = 0>
+    void logOct(const T value) noexcept;
 
   private:
+    // AXIVION Next Construct AutosarC++19_03-A3.9.1 : Not used as an integer but as actual character
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : C-style array is used to acquire size of the array safely. Safe
+    // access is guaranteed since the char array is not accessed but only the size is obtained
     template <uint32_t N>
-    // NOLINTJUSTIFICATION safe access is guaranteed since the char array is not accessed but only the size is obtained
     // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
-    inline static constexpr uint32_t bufferSize(const char (&)[N]) noexcept;
+    static constexpr uint32_t bufferSize(const char (&)[N]) noexcept;
 
     template <typename T>
-    inline void unused(T&&) const noexcept;
+    static constexpr void unused(T&&) noexcept;
 
+    // AXIVION Next Construct AutosarC++19_03-A3.9.1 : Not used as an integer but format string literal
     template <typename T>
-    inline void logArithmetic(const T value, const char* format) noexcept;
+    static void logArithmetic(const T value, const char* format) noexcept;
```

Comment:
We do not gain anything by this since the value is used for `snprintf` which expects a `const char*`. In addition, this is an implementation detail and it is ensured that only valid format strings are passed.